state/etcdstatedriver.go

```diff
@@ -53,19 +53,28 @@ type EtcdStateDriver struct {
 // Init the driver with a core.Config.
 func (d *EtcdStateDriver) Init(instInfo *core.InstanceInfo) error {
 	var err error
+	var endpoint *url.URL
 
-	if instInfo == nil || !strings.Contains(instInfo.DbURL, "etcd://") {
-		return errors.New("invalid etcd config")
+	if instInfo == nil || instInfo.DbURL == "" {
+		return errors.New("no Etcd config found")
 	}
-
-	etcdURL := strings.Replace(instInfo.DbURL, "etcd://", "http://", 1)
+	endpoint, err = url.Parse(instInfo.DbURL)
+	if err != nil {
+		return err
+	}
+	if endpoint.Scheme == "etcd" {
+		endpoint.Scheme = "http"
+	} else if endpoint.Scheme != "http" && endpoint.Scheme != "https" {
+		return core.Errorf("invalid Etcd URL scheme %q", endpoint.Scheme)
+	}
+	// TODO: support multi-endpoints
```

Comment:
You can `strings.Split()` on a comma, iterate over the result, validate the URLs, and append to the array you set as `Endpoints`

Reply:
the reason it still only supports single endpoint is because netmaster also shares the same part of codes, i need to make netmaster using new way then tweak these codes, it's on my list